src/convertToObject.js

```diff
@@ -6,7 +6,19 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const result = {};
+  const editWords = (words) => words.trim();
+  const editArray = sourceString
+    .split(';')
+    .filter((el) => el !== '' && el !== '  ;')
+    .map((arr) => arr.split(':'))
+    .map((arr) => arr.map(editWords));
```

Comment:
[CODE KNOWLEDGE] - There is no need to use 'map' twice. You can perform the 'split' and 'trim' operations in a single 'map' call.